pkg/logging/cri_logger.go

```diff
@@ -113,56 +113,62 @@ func ReadLogs(ctx context.Context, opts *LogViewOptions, stdout, stderr io.Write
 	writer := newLogWriter(stdout, stderr, opts)
 	msg := &logMessage{}
 	for {
-		if stop || (limitedMode && limitedNum == 0) {
-			logrus.Debugf("Finished parsing log file, path; %s", logPath)
+		select {
+		case <-stopChannel:
+			logrus.Debugf("received stop signal while reading cri logfile, returning")
 			return nil
-		}
-		l, err := r.ReadBytes(eol[0])
-		if err != nil {
-			if err != io.EOF { // This is an real error
-				return fmt.Errorf("failed to read log file %q: %v", logPath, err)
+		default:
+			if stop || (limitedMode && limitedNum == 0) {
+				logrus.Debugf("Finished parsing log file, path; %s", logPath)
+				return nil
 			}
-			if opts.Follow {
+			l, err := r.ReadBytes(eol[0])
+			if err != nil {
+				if err != io.EOF { // This is an real error
+					return fmt.Errorf("failed to read log file %q: %v", logPath, err)
+				}
+				if opts.Follow {
+
+					// Reset seek so that if this is an incomplete line,
+					// it will be read again.
+					if _, err := f.Seek(-int64(len(l)), io.SeekCurrent); err != nil {
+						return fmt.Errorf("failed to reset seek in log file %q: %v", logPath, err)
+					}
 
-				// Reset seek so that if this is an incomplete line,
-				// it will be read again.
-				if _, err := f.Seek(-int64(len(l)), io.SeekCurrent); err != nil {
-					return fmt.Errorf("failed to reset seek in log file %q: %v", logPath, err)
+					// If the container exited consume data until the next EOF
+					continue
 				}
+				// Should stop after writing the remaining content.
+				stop = true
+				if len(l) == 0 {
+					continue
+				}
+				logrus.Debugf("Incomplete line in log file, path: %s line: %s", logPath, l)
```

Comment:
nit: add a separator between `path` and `line` (e.g. `path: %s, line: %s` or `;`). otherwise the log may look confusing (e.g. `path: /a/b/c/d/fff line: datatatata`)

Reply:
updated